### PR TITLE
GG-18641: Statistics obsolescence

### PR DESCRIPTION
**Pull request by Berkof**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/config/StatisticsColumnConfiguration.java [hunk 1]
```diff
@@ -36,24 +36,37 @@
...
         this.name = name;
         this.ver = ver;
         this.tombstone = tombstone;
     }
 
-    /** */
+    /**
+     * Constructor.
+     *
+     * @param cfg Base staticsitcs column configuration
```
Comment (tledkov-gridgain) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/config/StatisticsColumnConfiguration.java:65]: Typo: 'statistics'

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManager.java [hunk 2]
```diff
@@ -24,9 +27,17 @@
     /**
      * Gather object statistics.
      *
-     * @param targets Target to gather statistics by.
+     * @param targets Target to params map to gather statistics by.
      * @throws IgniteCheckedException  Throws in case of errors.
      */
+    public void collectStatistics(StatisticsObjectConfiguration... targets) throws IgniteCheckedException;
```
Comment (tledkov-gridgain) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManager.java:33]: I guess we have to drop old method collectStatistics(StatisticsTarget..).
Please run and fix test

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java [hunk 3]
```diff
@@ -511,13 +513,9 @@ private void processKillQueryCommand(SqlKillQueryCommand cmd) {
...
 
-        IgniteStatisticsManager statMgr = ctx.query().getIndexing().statsManager();
-
-        StatisticsTarget[] targets = cmd.targets().stream()
-            .map(t -> (t.schema() == null) ? new StatisticsTarget(cmd.schemaName(), t.obj(), t.columns()) : t)
-            .toArray(StatisticsTarget[]::new);
+        IgniteH2Indexing indexing = (IgniteH2Indexing)ctx.query().getIndexing();
 
-        statMgr.collectStatistics(targets);
+        indexing.statsManager().collectStatistics(cmd.targets().toArray(new StatisticsObjectConfiguration[0]));
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java:518]: StatisticTarget array or StatisticsObjectConfiguration array?

Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java:518]: Yep, configurations().

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsObsolescenceTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+        assertTrue(GridTestUtils.waitForCondition(() -> {
+            ObjectStatisticsImpl stat2 = (ObjectStatisticsImpl)statisticsMgr(0).getLocalStatistics(SMALL_KEY);
+
+            return stat2.rowCount() > stat1.rowCount();
+        }, 7000));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsObsolescenceTest.java:71]: Please rearrange tests and config/after methods

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsStorageTest.java [hunk 5]
```diff
@@ -35,13 +35,13 @@
      */
     @Test
     public void clearAllTest() throws Exception {
-        IgniteStatisticsManager statsMgr = grid(0).context().query().getIndexing().statsManager();
-        IgniteStatisticsRepository statsRepo = ((IgniteStatisticsManagerImpl) statsMgr).statisticsRepository();
+
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsStorageTest.java:38]: empty line, redundant Exception

**Pull request merged**